Merge net8.0 into main

mattleibow opened the pull request:
### Description of Change

Updates the main/SR 1 branch from net8.0.

dustin-wojciechowski committed:
Make Entitlements in Templates for MacCatalyst more consistent between debug and release configs. (#17833)

* Updated integration test to check for app-sandbox and network.client entitlements in debug config as well for maui-blazor.

* Added app-sandbox and network.client entitlements to both debug and release entitlement plists for maui-blazor and maui-mobile templates for MacCatalyst. get-task-allow is commented out for debug for maui-mobile but is not for maui-blazor.

* Removed unnecessary space

* Merged debug and release entitlement plists, edition conditionals in csproj

* Removed unnecessary items from csproj

mattleibow committed:
Revert "Work around the dotnet tool restore issues" (#17983)

Eilon committed:
Update Blazor Hybrid template to match ASP.NET Core Blazor Web template (#17946)

Fixes #16797

Eilon committed:
Enable <Nullable> in .NET MAUI project templates (#18019)

Fixes #17987

Eilon committed:
Set Nullable=enable in Control Samples projects (#18020)

This was to see if anything particularly unexpected happens due to making the templates have Nullable=enable by looking to see what happens in relatively large/complex projects. The conclusion is that it all seems to be just fine.

rmarinho committed:
Revert new packages names for iOS (#17980)

jonathanpeppers committed:
[android] update AOT profile for .NET 8 GA (#18061)

To used the very latest builds, I updated locally:

    <!-- dotnet/installer -->
    -<MicrosoftDotnetSdkInternalPackageVersion>8.0.100-rtm.23506.1</MicrosoftDotnetSdkInternalPackageVersion>
    +<MicrosoftDotnetSdkInternalPackageVersion>8.0.100-rtm.23513.4</MicrosoftDotnetSdkInternalPackageVersion>
    <!-- dotnet/runtime -->
    -<MicrosoftNETCoreAppRefPackageVersion>8.0.0-rtm.23509.5</MicrosoftNETCoreAppRefPackageVersion>
    +<MicrosoftNETCoreAppRefPackageVersion>8.0.0-rtm.23513.17</MicrosoftNETCoreAppRefPackageVersion>
    <!-- xamarin-android -->
    -<MicrosoftAndroidSdkWindowsPackageVersion>34.0.12</MicrosoftAndroidSdkWindowsPackageVersion>
    +<MicrosoftAndroidSdkWindowsPackageVersion>34.0.25</MicrosoftAndroidSdkWindowsPackageVersion>

Before updating the profile, an average of 10 runs on a Pixel 5 of a
`dotnet new maui` app:

    Average(ms): 557.9
    Std Err(ms): 1.73493515728975
    Std Dev(ms): 5.48634668973808

After updating the profile:

    Average(ms): 551.3
    Std Err(ms): 2.82075639973867
    Std Dev(ms): 8.92001494767058

These are even better numbers than we got in 2d8cf282. It's possible we
slightly beat .NET 7 now, I'll test.

github-actions[bot] committed:
[ci] Fix api leves on tests (#18133)

Co-authored-by: Rui Marinho <me@ruimarinho.net>

jsuarezruiz committed:
[Windows] Fix wrong Layout from hidden CollectionView  (#17659)

* Fix wrong Layout from hidden CollectionView

* Applied some changes

StephaneDelcroix committed:
[X] connect the right method to the event handler (#18043)

- fixes #17950

StephaneDelcroix committed:
[iOS] Keep textcolor on characterspacing (#17926)

* add tests

* fix

emaf committed:
[WinUI] Fixes window blinking when resizing (#18155)

This bug was introduced by https://github.com/dotnet/maui/pull/17069,
but the underlying cause is a bug on WinUI that makes the window blink when
setting `NavigationView.IsPaneVisible = false` (https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/47131770).

After testing the samples attached on the issues the original commit fixed, this change does not seem to re-introduce those issues.

Fixes https://github.com/dotnet/maui/issues/17724

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/xamarin/xamarin-macios build 20231019.9 (#18157)

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8438 -> To Version 17.0.8445

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[net8.0] Update dependencies from xamarin/xamarin-macios (#18171)

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231019.52

Microsoft.iOS.Sdk
 From Version 17.0.8438 -> To Version 17.0.8446

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231019.52

Microsoft.tvOS.Sdk
 From Version 17.0.8438 -> To Version 17.0.8446

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231019.52

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8445 -> To Version 17.0.8446

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231019.52

Microsoft.macOS.Sdk
 From Version 14.0.8438 -> To Version 14.0.8446

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

rmarinho committed:
Update rtm versions (#18167)

mattleibow committed:
Disable the new code for now (#18198)

mattleibow committed:
[net8.0] Add permission for high-speed sensors (#18180)

StephaneDelcroix committed:
[X] revert #5611 (#17733)

As reported in #17461, OnPlatform simplification cause some issues
- while used as a Resource (#17461, unable to unit test for now)
- doesn't type convert. On nodes shouldn't be replaced by the Value, but
  an element node (<On Platform="..." Value="Red" /> should generate
<Color x:Key="foo">Red</Color>. right now it only generated "Red"

- fixes #17461

jknaudt21 committed:
[iOS] Fix downscaled images not occupying correct space (#17120)

dotnet-maestro[bot] committed:
[net8.0] Update dependencies from xamarin/xamarin-macios (#18191)

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.16

Microsoft.iOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8447

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.16

Microsoft.tvOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8447

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.16

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8446 -> To Version 17.0.8447

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.16

Microsoft.macOS.Sdk
 From Version 14.0.8446 -> To Version 14.0.8447

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.17

Microsoft.iOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8448

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.17

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8446 -> To Version 17.0.8448

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.17

Microsoft.macOS.Sdk
 From Version 14.0.8446 -> To Version 14.0.8448

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.17

Microsoft.tvOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8448

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.28

Microsoft.iOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8450

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.28

Microsoft.tvOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8450

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.28

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8446 -> To Version 17.0.8450

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.28

Microsoft.macOS.Sdk
 From Version 14.0.8446 -> To Version 14.0.8450

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.29

Microsoft.tvOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8451

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.29

Microsoft.macOS.Sdk
 From Version 14.0.8446 -> To Version 14.0.8451

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.29

Microsoft.iOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8451

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.29

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8446 -> To Version 17.0.8451

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/aspnetcore build 20231020.10 (#18224)

Microsoft.AspNetCore.Authentication.Facebook , Microsoft.AspNetCore.Authentication.Google , Microsoft.AspNetCore.Authentication.MicrosoftAccount , Microsoft.AspNetCore.Authorization , Microsoft.AspNetCore.Components , Microsoft.AspNetCore.Components.Analyzers , Microsoft.AspNetCore.Components.Forms , Microsoft.AspNetCore.Components.Web , Microsoft.AspNetCore.Components.WebView , Microsoft.AspNetCore.Metadata , Microsoft.JSInterop
 From Version 8.0.0-rtm.23518.6 -> To Version 8.0.0-rtm.23520.10

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

mattleibow committed:
Rework the Label mappers for FormattedText (#18105)

The current implementation did not recreate the FormattedText spans when certain properties changed: Font and TextColor

StephaneDelcroix committed:
[C] test for #17776 (#17796)

rmarinho committed:
Update net8 rtm versions (#18247)

dotnet-maestro[bot] committed:
[net8.0] Update dependencies from xamarin/xamarin-macios (#18248)

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.1

Microsoft.tvOS.Sdk
 From Version 17.0.8451 -> To Version 17.0.8452

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.1

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8451 -> To Version 17.0.8452

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.1

Microsoft.macOS.Sdk
 From Version 14.0.8451 -> To Version 14.0.8452

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.1

Microsoft.iOS.Sdk
 From Version 17.0.8451 -> To Version 17.0.8452

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.9

Microsoft.iOS.Sdk
 From Version 17.0.8451 -> To Version 17.0.8453

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.9

Microsoft.tvOS.Sdk
 From Version 17.0.8451 -> To Version 17.0.8453

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.9

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8451 -> To Version 17.0.8453

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.9

Microsoft.macOS.Sdk
 From Version 14.0.8451 -> To Version 14.0.8453

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

mattleibow committed:
Add NuGet package information (#18197)

github-actions[bot] committed:
Correctly expand rows/columns when grid padding (#18276)

When expanding star definitions, the targetSize
parameter contains the padding (which is
removed). However, the calculation for the
the GridMinimum[Width|Height]() also has the
padding which also needs to be removed.

Co-authored-by: Matthew Leibowitz <mattleibow@live.com>

mattleibow committed:
Switch to RC 2 SDK and packages

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/aspnetcore build 20231023.8 (#18286)

Microsoft.AspNetCore.Authentication.Facebook , Microsoft.AspNetCore.Authentication.Google , Microsoft.AspNetCore.Authentication.MicrosoftAccount , Microsoft.AspNetCore.Authorization , Microsoft.AspNetCore.Components , Microsoft.AspNetCore.Components.Analyzers , Microsoft.AspNetCore.Components.Forms , Microsoft.AspNetCore.Components.Web , Microsoft.AspNetCore.Components.WebView , Microsoft.AspNetCore.Metadata , Microsoft.JSInterop
 From Version 8.0.0-rtm.23520.10 -> To Version 8.0.0-rtm.23523.8

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[net8.0] Update dependencies from xamarin/xamarin-macios (#18281)

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.3

Microsoft.tvOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8454

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.3

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8453 -> To Version 17.0.8454

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.3

Microsoft.macOS.Sdk
 From Version 14.0.8453 -> To Version 14.0.8454

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.3

Microsoft.iOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8454

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.5

Microsoft.tvOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8455

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.5

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8453 -> To Version 17.0.8455

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.5

Microsoft.macOS.Sdk
 From Version 14.0.8453 -> To Version 14.0.8455

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.5

Microsoft.iOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8455

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.19

Microsoft.tvOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8456

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.19

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8453 -> To Version 17.0.8456

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.19

Microsoft.macOS.Sdk
 From Version 14.0.8453 -> To Version 14.0.8456

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.19

Microsoft.iOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8456

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.20

Microsoft.iOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8457

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.20

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8453 -> To Version 17.0.8457

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.20

Microsoft.macOS.Sdk
 From Version 14.0.8453 -> To Version 14.0.8457

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.20

Microsoft.tvOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8457

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.21

Microsoft.iOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8458

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.21

Microsoft.tvOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8458

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.21

Microsoft.macOS.Sdk
 From Version 14.0.8453 -> To Version 14.0.8458

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.21

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8453 -> To Version 17.0.8458

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

tj-devel709 committed:
Allow iOS Keyboard Scrolling to be turned off in Lifecycle Events (#18266)

* Add spacing

* Use a better implementation

PureWeen committed:
Setup legacy mappers (#18087)

* Route Controls Mapper to Core Mapper

* Update ControlsMapper.cs

* - fix test

dustin-wojciechowski committed:
[Templates] Remove get-task-allow entitlement from Templates (#18272)

* Removed get-task-allow entitlement from templates as they do not affect usability of developer tools in safari

* Updated CheckEntitlements unit test to remove check for get-task-allow

github-actions[bot] committed:
[release/8.0.1xx-rc2.2] Allow iOS Keyboard Scrolling to be turned off in Lifecycle Events (#18308)

* Add spacing

* Use a better implementation

---------

Co-authored-by: tj-devel709 <tj.devel709@gmail.com>

github-actions[bot] committed:
[release/8.0.1xx-rc2.2] Setup legacy mappers (#18309)

* Route Controls Mapper to Core Mapper

* Update ControlsMapper.cs

* - fix test

---------

Co-authored-by: Shane Neuville <shneuvil@microsoft.com>

hartez committed:
Ensure that viewport calculations for iOS ScrollView account for content insets (#18277)

* Ensure that viewport calculations for iOS ScrollView account for content insets
Fixes #17224

* Update baseline image

github-actions[bot] committed:
[release/8.0.1xx-rc2.2] Make sure to account for a null AppWindow (#18313)

* Make sure to account for a null AppWindow

* Update Issue17490.cs

* - fix banned api

* - fix automationid

---------

Co-authored-by: Shane Neuville <shneuvil@microsoft.com>

github-actions[bot] committed:
[release/8.0.1xx-rc2.2] API Doc improvements to fix online browser warnings (#18311)

* Update View.cs

* Update Element.cs

* More fixes

* Update Page.cs

* Remove dead links

* More fixes

* Remove developer.xamarin.com references public doc

* Remove developer.xamarin.com refs from comments

* Remove xamarin-forms references mostly

* Update VisualElement.xml

* Remove unused XML doc files

* OK maybe not VisualElement

---------

Co-authored-by: Gerald Versluis <gerald.versluis@microsoft.com>

github-actions[bot] committed:
[release/8.0.1xx-rc2.2] Limit star expansion when measuring/arranging at constrained sizes (#18314)

* Limit star expansion when measuring/arranging at constrained sizes Fixes #17125

* Clean up and expand tests a bit

---------

Co-authored-by: E.Z. Hart <hartez@gmail.com>

jknaudt21 committed:
Ensure templates create a proper AppID (#18322)

github-actions[bot] committed:
[release/8.0.1xx-rc2.2] [Templates] Remove get-task-allow entitlement from Templates (#18323)

* Removed get-task-allow entitlement from templates as they do not affect usability of developer tools in safari

* Updated CheckEntitlements unit test to remove check for get-task-allow

---------

Co-authored-by: Dustin Wojciechowski <dustin.wojciechowski@microsoft.com>

github-actions[bot] committed:
[release/8.0.1xx-rc2.2] Ensure that viewport calculations for iOS ScrollView account for content insets (#18321)

* Ensure that viewport calculations for iOS ScrollView account for content insets Fixes #17224

* Update baseline image

---------

Co-authored-by: E.Z. Hart <hartez@gmail.com>

PureWeen committed:
Bump (#18326)

mattleibow committed:
Update .NET 7 MAUI Version (#18329)

* Update .NET 7 MAUI Version

* Update Versions.props

dotnet-maestro[bot] committed:
[net8.0] Update dependencies from xamarin/xamarin-macios (#18307)

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.27

Microsoft.iOS.Sdk
 From Version 17.0.8458 -> To Version 17.0.8459

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.27

Microsoft.tvOS.Sdk
 From Version 17.0.8458 -> To Version 17.0.8459

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.27

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8458 -> To Version 17.0.8459

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.27

Microsoft.macOS.Sdk
 From Version 14.0.8458 -> To Version 14.0.8459

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.46

Microsoft.iOS.Sdk
 From Version 17.0.8458 -> To Version 17.0.8460

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.46

Microsoft.tvOS.Sdk
 From Version 17.0.8458 -> To Version 17.0.8460

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.46

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8458 -> To Version 17.0.8460

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.46

Microsoft.macOS.Sdk
 From Version 14.0.8458 -> To Version 14.0.8460

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.47

Microsoft.tvOS.Sdk
 From Version 17.0.8458 -> To Version 17.0.8461

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.47

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8458 -> To Version 17.0.8461

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.47

Microsoft.macOS.Sdk
 From Version 14.0.8458 -> To Version 14.0.8461

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.47

Microsoft.iOS.Sdk
 From Version 17.0.8458 -> To Version 17.0.8461

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231025.1

Microsoft.tvOS.Sdk
 From Version 17.0.8458 -> To Version 17.0.8462

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231025.1

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8458 -> To Version 17.0.8462

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231025.1

Microsoft.macOS.Sdk
 From Version 14.0.8458 -> To Version 14.0.8462

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231025.1

Microsoft.iOS.Sdk
 From Version 17.0.8458 -> To Version 17.0.8462

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231025.2

Microsoft.iOS.Sdk
 From Version 17.0.8458 -> To Version 17.0.8463

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231025.2

Microsoft.tvOS.Sdk
 From Version 17.0.8458 -> To Version 17.0.8463

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231025.2

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8458 -> To Version 17.0.8463

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231025.2

Microsoft.macOS.Sdk
 From Version 14.0.8458 -> To Version 14.0.8463

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

mattleibow committed:
Merge branch 'release/8.0.1xx-rc2.2' into dev/merge-rc2.2-into-net8

jknaudt21 committed:
[Templates] Fix template typo (#18166)

mattleibow committed:
Merge RC 2.2 into net8.0 (#18351)

### Description of Change

Upstreaming changes.

mattleibow committed:
Support finding nearest TFM to collect assets from (#18068)

emaf committed:
Partially revert "[Windows] Update CollectionView changing ItemsLayout  (#14532)" (#18356)

This partially reverts commit d569287ee81d6e5fd0f89981befd58c25c70da77,
keeping the AdaptiveCollectionView sample.

The reverted change was switching to only use FormsGridView as backing
control for CollectionView what causes issues like
https://github.com/dotnet/maui/issues/16234.

jsuarezruiz reviewed (commented):
The build is failing:

```
D:\a\_work\1\s\src\Controls\tests\UITests\Tests\Issues\Issue17400.cs(1,22): error CS0234: The type or namespace name 'Appium' does not exist in the namespace 'Microsoft.Maui' (are you missing an assembly reference?) [D:\a\_work\1\s\src\Controls\tests\UITests\Controls.AppiumTests.csproj::TargetFramework=net8.0]
D:\a\_work\1\s\src\Controls\tests\UITests\Tests\Issues\Issue17400.cs(8,21): error CS0246: The type or namespace name 'TestDevice' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Controls\tests\UITests\Controls.AppiumTests.csproj::TargetFramework=net8.0]
```

The test Issue17400 just require a small change in the namespaces:
```
using NUnit.Framework;
using UITest.Appium;
using UITest.Core;
```

hartez committed:
Make Android respect power saving/disabled animations (#17287)

* Re-enable disabled animation tests;
Make Android respect power saving/disabled animations;
Watch animation duration if API >= 33

* Nicer API for running single-thread tests; fix intermittent crashes

* Fix shipped nonsense

* Fix shipped nonsense

* Address Matt's comments

rachelkang committed:
.NET 8 template tweaks (#18379)

* Dave's template changes

* Rachel small tweaks part 1

* Remove png, swap in svg version 1

* Fix spacing

* Replace image with 1200px height version

* Change to png again

Eilon committed:
Fix project name substitution in Blazor template nav menu (#18388)

Fixes #18382

PureWeen committed:
Fix Device Info Test API 33 (#18390)

* Fix device model error api 33

* - add sdk_gphone64_x86_64

StephaneDelcroix committed:
[C] clear _targetObject (#18398)

Avoid NRE on reapplying.

- fixes #18123

rmarinho committed:
[net8.0] Update stable branding (#18384)

* Update dependencies from https://github.com/dotnet/aspnetcore build 20231024.15

Microsoft.AspNetCore.Authentication.Facebook , Microsoft.AspNetCore.Authentication.Google , Microsoft.AspNetCore.Authentication.MicrosoftAccount , Microsoft.AspNetCore.Authorization , Microsoft.AspNetCore.Components , Microsoft.AspNetCore.Components.Analyzers , Microsoft.AspNetCore.Components.Forms , Microsoft.AspNetCore.Components.Web , Microsoft.AspNetCore.Components.WebView , Microsoft.AspNetCore.Metadata , Microsoft.JSInterop
 From Version 8.0.0-rtm.23523.8 -> To Version 8.0.0-rtm.23524.15

* Bump other stuff to stable branding

* Bump to xamarin/xamarin-android/release/8.0.1xx@b48acafb 34.0.39 (#18381)

Changes: https://github.com/xamarin/xamarin-android/compare/ad4af58b...b48acafb
Changes: https://github.com/dotnet/runtime/compare/11ad607e...59edaad4

This also updates to 8.0.0 stable branded dotnet/runtime and dotnet/emsdk.

`.Transport` was renamed from dotnet/emsdk, so I manually ran:

    > darc update-dependencies --id 199105
    Looking up build with BAR id 199105
    Updating 'Microsoft.Android.Sdk.Windows': '34.0.36' => '34.0.39' (from build '8.0.1xx-b48acafb42921772e76464917d3c0900a009d680-1' of 'https://github.com/xamarin/xamarin-android')
    Checking for coherency updates...
    Using 'Strict' coherency mode. If this fails, a second attempt utilizing 'Legacy' Coherency mode will be made.
    Updating 'Microsoft.Dotnet.Sdk.Internal': '8.0.100-rtm.23522.1' => '8.0.100-rtm.23518.33' to ensure coherency with Microsoft.Android.Sdk.Windows@34.0.39
    Updating 'Microsoft.NETCore.App.Ref': '8.0.0-rtm.23520.16' => '8.0.0' to ensure coherency with Microsoft.Android.Sdk.Windows@34.0.39
    Updating 'Microsoft.NET.Workload.Emscripten.Current.Manifest-8.0.100': '8.0.0-rtm.23511.3' => '8.0.0' to ensure coherency with Microsoft.NETCore.App.Ref@8.0.0-rtm.23520.16
    Local dependencies updated based on build with BAR id 199105 (8.0.1xx-b48acafb42921772e76464917d3c0900a009d680-1 from https://github.com/xamarin/xamarin-android@release/8.0.1xx)

Then I chose the higher version of dotnet/installer, which was already
present here.

* s!

* .NET 6 and .NET 7 servicing feeds

Got these from: https://github.com/xamarin/xamarin-android/commit/f63ac29fb3e48bfb5785e9fad03a64f68f7fc750

* Updating 'Microsoft.Android.Sdk.Windows': '34.0.39' => '34.0.40'

* Fix NRT warnings

* [iOS] Use new API for UISceneSessionActivation

* [iOS] Disable CA1422 for TraitCollectionDidChange is obsolete on iOS17

* [iOS] Disable warning on TraitCollectionDidChange on old renderers

* [tests] Fix Assetion on iOS

* Still missing some TraitCollectionDidChange

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Jonathan Peppers <jonathan.peppers@microsoft.com>

rmarinho committed:
[net8.0] Bump xharness for rtm and bring iOS17 tests (#18353)

* Bum xharness for rtm

* Fix ios cake

* Add ios17

* Update device-tests.yml

* [tests] Try fix test crash on iOS17

* [tests] Add code to launch on device

* Skip test on iOS17

* Try fix UITests

* Pass device name

* Fix run uitests with correct dotnet

* Update Compatibility.ControlGallery.iOS.csproj

* Fix this

* Ignore visual tests on iPhone 15 devices (#18407)

* Fix TapTwoPlacesQuickly on IOS17

---------

Co-authored-by: Shane Neuville <shneuvil@microsoft.com>
Co-authored-by: Bret Johnson <bret.johnson@microsoft.com>

mattleibow committed:
Merge branch 'net8.0' into dev/merge-net8.0-into-main

mattleibow commented:
This needs to be recreated.
